utils/land-patterns/pose-arrangement.stanza

```diff
@@ -0,0 +1,127 @@
+defpackage ocdb/land-patterns/pose-arrangement:
+  import core
+  import collections
+  import math
+  import jitx
+  
+defstruct Affine:
+  matrix: [
+    [Double, Double, Double],
+    [Double, Double, Double],
+    [Double, Double, Double]
+  ]
+
+defn Affine () -> Affine:
+  Affine([
+    [1.0, 0.0, 0.0]
+    [0.0, 1.0, 0.0]
+    [0.0, 0.0, 1.0]
+  ])
+
+defn zero () -> Affine:
+    Affine([
+    [0.0, 0.0, 0.0]
+    [0.0, 0.0, 0.0]
+    [0.0, 0.0, 0.0]
+  ])
+
+defn times (a:Affine, b:Affine) -> Affine:
+  val [A, B] = [matrix(a), matrix(b)]
+  val C = Array<Array<Double>>(3)
+  
+  for i in 0 to 3 do:
+    C[i] = Array<Double>(3)
+    for j in 0 to 3 do: 
+      C[i][j] = 0.0
+      for k in 0 to 3 do: 
+        C[i][j] = C[i][j] + A[i][k] * B[k][j]
+  Affine([
+    [C[0][0], C[0][1], C[0][2]]
+    [C[1][0], C[1][1], C[1][2]]
+    [C[2][0], C[2][1], C[2][2]]
+  ])
+
+defn scale (x:Double, y:Double) -> Affine:
+  Affine([
+    [x, 0.0, 0.0]
+    [0.0, y, 0.0]
+    [0.0, 0.0, 1.0]
+  ])
+
+defn flip-x () -> Affine:
+  scale(-1.0, 1.0)
+
+defn flip-y () -> Affine:
+  scale(1.0, -1.0)
+
+defn translate (x:Double, y:Double) -> Affine:
+  Affine([
+    [1.0, 0.0, x]
+    [0.0, 1.0, y]
+    [0.0, 0.0, 1.0]
+  ])
+ 
+defn rotation (degrees:Double) -> Affine:
+  val theta = PI * degrees / 180.0
+  Affine([
+    [cos(theta), -1.0 * sin(theta), 0.0],
+    [sin(theta), cos(theta), 0.0],
+    [0.0, 0.0, 1.0]
+  ])
+
+defn times (p:Point, a:Affine) -> Point:
+  val p1 = Array<Double>(3)
+  p1[0] = x(p)
+  p1[1] = y(p)
+  p1[2] = 1.0
+
+  var p2 = Array<Double>(2)
+  p2[0] = 0.0
+  p2[1] = 0.0
+  p2[2] = 0.0
+
+  val A = matrix(a)
+  for i in 0 to 3 do:
+    for j in 0 to 3 do:
+      p2[i] = p2[i] + p1[j] * A[i][j]
+  Point(p2[0], p2[1])
+
+defn times (p:Pose, a:Affine):
+  sub-center(p, center(p) * a) 
+
+;
+public defstruct PoseArrangement: 
+  locations: Seqable<Pose>,
+
+defn times (a:Affine, p:PoseArrangement) -> PoseArrangement:
+  times(p, a)
+
+defn times (p:PoseArrangement, a:Affine) -> PoseArrangement:
+  PoseArrangement(seq({_ * a}, locations(p)))
+
+public defn translated (p:PoseArrangement, x:Double, y:Double) -> PoseArrangement:
```

Comment:
Maybe that's too much but why not `public defn plus (p:PoseArrangement, xy:[Double, Double]) -> PoseArrangement :`?

or `xy:Array<Double>(2)` if it is supported

Reply:
The root reason is because in frameworks where I have used these operations, they use the explicit name of the transform (`scale`, `shear`, `translate`, `rotate`, etc). I also find the explicitness more readable than operator overloading.

The difference between the implementation of `times` in here is that under the hood, it is just matrix multiplication. However that is only exposed internally, the functions are not public.

Reply:
why not having both?

```
public defn plus (p:PoseArrangement, xy:[Double, Double]) -> PoseArrangement :
   translated(p, x, y)
```